src/main/java/dinkplugin/notifiers/LeaguesNotifier.java

```diff
@@ -0,0 +1,290 @@
+     */
+    @VisibleForTesting
+    static final @Varp int POINTS_EARNED_ID = 2614;
+
+    /**
+     * @see <a href="https://github.com/Joshua-F/cs2-scripts/blob/fa31b06ec5a9f6636bf9b9d5cbffbb71df022d06/scripts/%5Bproc%2Cleague_areas_draw_interface%5D.cs2#L28-L55">CS2 Reference</a>
+     */
+    @VisibleForTesting
+    static final @Varbit int FIVE_AREAS = 10666, FOUR_AREAS = 10665, THREE_AREAS = 10664, TWO_AREAS = 10663;
+
+    /**
+     * Trophy name by the required points, in a binary search tree.
+     *
+     * @see <a href="https://oldschool.runescape.wiki/w/Trailblazer_Reloaded_League#Trophies">Wiki Reference</a>
+     * @see <a href="https://github.com/Joshua-F/cs2-scripts/blob/fa31b06ec5a9f6636bf9b9d5cbffbb71df022d06/scripts/%5Bproc%2Cscript731%5D.cs2#L3">CS2 Reference</a>
+     */
+    private static final NavigableMap<Integer, String> TROPHY_BY_POINTS;
+
+    /**
+     * Mapping of each relic name to the tier (1-8).
+     *
+     * @see <a href="https://oldschool.runescape.wiki/w/Trailblazer_Reloaded_League/Relics">Wiki Reference</a>
+     */
+    private static final Map<String, LeagueRelicTier> TIER_BY_RELIC;
+
+    /**
+     * Mapping of the number of tasks required to unlock an area to the area index (0-3).
+     *
+     * @see <a href="https://oldschool.runescape.wiki/w/Trailblazer_Reloaded_League/Areas">Wiki reference</a>
+     */
+    private static final NavigableMap<Integer, Integer> AREA_BY_TASKS;
+
+    @Override
+    public boolean isEnabled() {
+        return config.notifyLeagues() &&
+            client.getWorldType().contains(WorldType.SEASONAL) &&
+            settingsManager.isNamePermitted(client.getLocalPlayer().getName());
+    }
+
+    @Override
+    protected String getWebhookUrl() {
+        return config.leaguesWebhook();
+    }
+
+    public void onGameMessage(String message) {
+        if (!isEnabled()) {
+            return;
+        }
+        if (message.startsWith(AREA_UNLOCK_PREFIX)) {
+            if (config.leaguesAreaUnlock()) {
+                String area = message.substring(AREA_UNLOCK_PREFIX.length(), message.length() - 1);
+                notifyAreaUnlock(area);
+            }
+        } else if (message.startsWith(RELIC_UNLOCK_PREFIX)) {
+            if (config.leaguesRelicUnlock()) {
+                String relic = message.substring(RELIC_UNLOCK_PREFIX.length(), message.length() - 1);
+                notifyRelicUnlock(relic);
+            }
+        } else if (config.leaguesTaskCompletion()) {
+            LeagueTaskDifficulty tier;
+            Matcher matcher = TASK_REGEX.matcher(message);
+            if (matcher.find() &&
+                (tier = LeagueTaskDifficulty.TIER_BY_LOWER_NAME.get(matcher.group("tier"))) != null &&
+                tier.ordinal() >= config.leaguesTaskMinTier().ordinal()) {
+                notifyTaskCompletion(tier, matcher.group("task"));
+            }
```

Comment:
Could the tier-logic be scoped so the assignment doesn't happen inside the if-statement?
e.g.
```java
if (matcher.find()) {
  tier = ...;
  if (tier != null & tier.ordinal...) {
    notifyTaskCompletion
  }
}
```